imap/search_xapian.c

```diff
@@ -358,31 +358,29 @@ static int xapstat(const char *path)
 {
     struct stat sbuf;
     int r;
-
-    /* is there a glass file? */
-    char *glass = strconcat(path, "/iamglass", (char *)NULL);
-    r = stat(glass, &sbuf);
-    free(glass);
-
-    /* zero byte file is the same as no database */
-    if (!r && !sbuf.st_size) {
-         r = -1;
-         errno = ENOENT;
-    }
-    if (!r) return 0;
-
-    /* check for old chert file */
-    char *chert = strconcat(path, "/iamchert", (char *)NULL);
-    r = stat(chert, &sbuf);
-    free(chert);
-
-    /* zero byte file is the same as no database */
-    if (!r && !sbuf.st_size) {
-         r = -1;
-         errno = ENOENT;
+    int i;
+    const char *known_files[] = {
+        "cyrus.indexed.db",
+        "iamglass",
+        "position.glass",
+        "postlist.glass",
+        "termlist.glass",
+        NULL
+    };
+
+    for (i = 0; known_files[i]; i++) {
+        char *fname = strconcat(path, "/", known_files[i], (char *)NULL);
```

Comment:
Given that we expect the majority of users to have all these files the cost of calling `malloc/free` five times looks unnecessary expensive, given that we could achieve the same by using a single `struct buf` for the building the file names in the loop.

But since the cost of calling `stat` is likely to make the cost of `malloc/free` look small in comparison, there's no big need for this optimization.

Reply:
Yeah, I don't think that it's anywhere near important compared to all the other costs. Not only stat but the locking cost per user - we have only a few files per user.